### PR TITLE
Switch to typescript

### PR DESCRIPTION
**Pull request by 6zz**
use ts-node runtime to run mocha tests.

update is breaking change: project will be switched to typescript from now on.  Hence changing the major pkg version

### CODE REVIEW

File: test/28.t-hello.test.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { assert } from 'chai';
+import { hello } from '../questions/28.t-hello';
+
+describe('Typescript Support tests', function() {
+    it('should run typescript code', function() {
+        assert.equal(hello('mate'), 'hellow mate');
+    })
+})
```
Comment (jasonly) [inline test/28.t-hello.test.ts:8]:
```suggestion
})

```

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "compilerOptions": {
+        "types": ["mocha"]
+    }
+}
```
Comment (jasonly) [inline tsconfig.json:5]:
```suggestion
}

```

File: questions/28.t-hello.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function hello(name: string): string {
+    return `hellow ${name}`;    
+}
```
Comment (jasonly) [inline questions/28.t-hello.ts:3]:
```suggestion
}

```

File: questions/28.t-hello.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function hello(name: string): string {
+    return `hellow ${name}`;
```
Comment (jasonly) [inline questions/28.t-hello.ts:2]:
```suggestion
    return `hello ${name}`;
```

Oh was the `w` intentional?

File: test/28.t-hello.test.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { assert } from 'chai';
+import { hello } from '../questions/28.t-hello';
+
+describe('Typescript Support tests', function() {
+    it('should run typescript code', function() {
+        assert.equal(hello('mate'), 'hellow mate');
```
Comment (jasonly) [inline test/28.t-hello.test.ts:6]:
```suggestion
        assert.equal(hello('mate'), 'hello mate');
```

**Comment by jasonly** [comment]
Omg thank you for doing this!

File: questions/28.t-hello.ts (hunk 4 above)
Comment (6zz) [reply questions/28.t-hello.ts:2]: I could say it's intended. 😆. But it really is a typo. 😝

**Comment by 6zz** [comment]
I have moved all the js files into archive.  unless we want to make it an exercise to convert previous questions/tests to typescript each week.  it could actually be good practice to get good at typescript.

File: questions/28.t-hello.ts (hunk 4 above)
Comment (jasonly) [reply questions/28.t-hello.ts]: hahahaha!

**Comment by jasonly** [comment]
>I have moved all the js files into archive. unless we want to make it an exercise to convert previous questions/tests to typescript each week. it could actually be good practice to get good at typescript.

Ohh that actually sounds like a really good idea! I wanna do that!

**Review by jasonly — APPROVED** [review]

**Comment by jasonly** [comment]
Thank you!!!

**Pull request merged**